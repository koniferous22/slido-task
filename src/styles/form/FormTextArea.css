form#case-add textarea {
	border-radius: 5px;
	resize: none;
}