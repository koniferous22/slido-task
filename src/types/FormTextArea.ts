export interface FormTextAreaProps {
	name: string;
	label: string;
	value: string;
	rows: number;
	errorFlag: boolean;
	onBlur: (s: string) => void;
	onChange: (s: string, e: React.ChangeEvent<HTMLTextAreaElement>) => void;
}
