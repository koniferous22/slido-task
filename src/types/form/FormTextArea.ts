import { Case } from 'types/common/Case'

export interface FormTextAreaProps {
	name: keyof Case;
	label: string;
	value: string;
	rows: number;
	errorFlag: boolean;
	onBlur: (field: keyof Case) => void;
	onChange: (field: keyof Case, value: string) => void;
}
