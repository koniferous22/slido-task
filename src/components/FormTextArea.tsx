import React from 'react'
import { FormTextAreaProps } from '../types/FormTextArea'

const FormTextArea: React.FC<FormTextAreaProps> = (props: FormTextAreaProps) => {
	const {name, label, errorFlag, onChange, onBlur, ...rest} = props
	return (
		<>
			{
				errorFlag && (
					<div className="error">
						Field is required
					</div>
				)
			}
			<label htmlFor={name}>
				{label + ': '}
			</label>
			<textarea
				{...rest}
				name={name}
				onChange={(event): void => onChange(name, event)}
				onBlur={(): void => onBlur(name)}
			/>			
		</>
	)
}

export default FormTextArea
