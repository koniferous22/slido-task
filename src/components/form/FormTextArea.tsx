import React from 'react'
import { FormTextAreaProps } from 'types/form/FormTextArea'

import 'styles/form/FormTextArea.css'

const FormTextArea: React.FC<FormTextAreaProps> = ({name, label, errorFlag, onChange, onBlur, ...textAreaProps}: FormTextAreaProps) => {
	return (
		<>
			{
				errorFlag && (
					<div className="error">
						Field is required
					</div>
				)
			}
			<label htmlFor={name}>
				{label + ': '}
			</label>
			<textarea
				{...textAreaProps}
				name={name}
				onChange={(event): void => onChange(name, event.target.value)}
				onBlur={(): void => onBlur(name)}
			/>			
		</>
	)
}

export default FormTextArea
