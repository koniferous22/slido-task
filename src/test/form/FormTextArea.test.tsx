import React from 'react'
import renderer from 'react-test-renderer'
import FormTextArea from 'components/form/FormTextArea'

it('FormTextArea renders correctly', () => {
	const tree = renderer
		.create(
			<FormTextArea
				name="input"
				label="Input"
				value="sample value"
				rows={5}
				errorFlag={false}
				onChange={(): void => {return;}}
				onBlur={(): void => {return;}}
			/>)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it('FormTextArea renders correctly with error', () => {
	const tree = renderer
		.create(
			<FormTextArea
				name="input"
				label="Input"
				value="sample value"
				rows={6}
				errorFlag={true}
				onChange={(): void => {return;}}
				onBlur={(): void => {return;}}
			/>)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
