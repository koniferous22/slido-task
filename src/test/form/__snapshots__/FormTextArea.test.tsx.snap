// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTextArea renders correctly 1`] = `
Array [
  <label
    htmlFor="input"
  >
    Input: 
  </label>,
  <textarea
    id="input"
    name="input"
    onBlur={[Function]}
    onChange={[Function]}
    rows={5}
    value="sample value"
  />,
]
`;

exports[`FormTextArea renders correctly with error 1`] = `
Array [
  <div
    className="error"
    id="error-input"
  >
    Field is required
  </div>,
  <label
    htmlFor="input"
  >
    Input: 
  </label>,
  <textarea
    id="input"
    name="input"
    onBlur={[Function]}
    onChange={[Function]}
    rows={6}
    value="sample value"
  />,
]
`;
